cts/cts.in

```diff
@@ -161,19 +161,36 @@ if [ x$cmd != x ]; then
     config="${CONFIG_DIR}/pacemaker"
     case $cmd in
 	trace-ls|tls)
-	    cluster-helper -g $cluster_name -- grep PCMK_trace_functions $config
+	    cluster-helper -g $cluster_name -- grep ^PCMK_trace_functions $config
 	    ;;
 	trace-add|tadd)
 	    echo "Adding $trace to PCMK_trace_functions"
-        sed_in_place_remotely "$config" "s/.*PCMK_trace_functions=/PCMK_trace_functions=$trace,/"
+        # Note that this only works if there's already a PCMK_trace_functions line.
+        # If there isn't one, create it with trace-set first.
+        sed_in_place_remotely "$config" "'/^PCMK_trace_functions=/ s/\"$/,$trace\"/'"
```

Comment:
technically the quotes are optional in sysconfig, and there could be whitespace before the variable name or between last quote and end of line (or even a # comment at the end). we don't have to handle every possible combination, but maybe whatever can be done in one command

Reply:
Fixed with an awful lot of regex work, which is hopefully commented well enough for whoever looks at this next.